scripts/s3_media_upload

```diff
@@ -394,11 +394,42 @@ def get_sqlite_conn(parser):
 
     return sqlite_conn
 
+def get_homeserver_db_conn(parser):
+    """Attempt to get a connection based on homeserver.yaml to Synapse's
+    database, or exit.
+    """
 
-def get_synapse_db_conn(parser):
+    try:
+        homeserver_yaml = yaml.safe_load(open("homeserver.yaml"))
```

Comment:
I think this should be wrapped in `with open(..) as f:` so we don't leak a file handle (unless `yaml.safe_load` definitely closes it?).

My only other concern: Is it definitely safe to assume the homeserver config is at `homeserver.yaml`? Should we accept some way of configuring it? I don't use this tool personally so I'm not familiar with it.
Of course, since the old `database.yaml` approach will still work, I guess it's unlikely to make things *worse* for anyone and we could add support to configure it later.

Reply:
> > unless yaml.safe_load definitely closes it?
> 
> I think that it does [closes it](https://github.com/yaml/pyyaml/blob/8cdff2c80573b8be8e8ad28929264a913a63aa33/lib/yaml/__init__.py#L83).

This is not correct. The `dispose`  method you link to is actually [`Parser.dispose`](https://github.com/yaml/pyyaml/blob/8cdff2c80573b8be8e8ad28929264a913a63aa33/lib/yaml/parser.py#L89-L92) (see `python -c 'import yaml; print(yaml.SafeLoader.dispose)'`) which only cleans up the parser's state.